.github/workflows/stable-sync.yml

```diff
@@ -119,9 +119,40 @@ jobs:
           # Check if branch exists remotely
           echo "Cleaning up github-tools"
           rm -rf github-tools
+          # Ensure github-tools is not staged or tracked
+          git rm -rf --cached github-tools 2>/dev/null || true
+          git clean -fd github-tools 2>/dev/null || true
+          # Commit the removal if it's staged
+          if git diff --cached --quiet; then
+            echo "No staged changes for github-tools"
+          else
+            git commit -m "Remove github-tools directory" --no-verify || true
+          fi
           BRANCH_NAME="stable-main-${{ inputs.semver-version }}"
           if git ls-remote --heads origin "$BRANCH_NAME" | grep -q "$BRANCH_NAME"; then
-            git pull --rebase
+            # Configure git to ignore submodules during operations
+            git config submodule.recurse false
+            git config diff.ignoreSubmodules all
+            
+            # Start the rebase
```

Comment:
I wonder whether we really need a rebase. Can we not just override the `origin/stable-main-${{ inputs.semver-version }}` branch with a git push force?